File: 412.py
# Author: Zoljargal Gantumur
# Runtime: 40 ms, faster than 80.98% of Python3 online submissions
# Memory Usage: 14.8 MB, less than 63.02% of Python3 online submissions


class Solution:
    def fizzBuzz(self, n: int) -> List[str]:
        ans = []
        for i in range(1,n+1):
            if i%3==0:
                ans.append("FizzBuzz") if i%5==0 else ans.append("Fizz")
            elif i%5==0:
                ans.append("FizzBuzz") if i%3==0 else ans.append("Buzz")
            else:
                ans.append(str(i))
        return ans
                

File: 412.cpp
/*
	Author: Zoljargal Gantumur
	Runtime: 0 ms, faster than 100.00% of C++ online submissions
	Memory Usage: 7.3 MB, less than 93.18% of C++ online submissions
*/

class Solution {
public:
    vector<string> fizzBuzz(int n) {
        vector<string> ans;
        ans.reserve(n);
        for(int i=1; i<=n; i++){
            if(i%3 == 0){
                if(i%5 == 0){
                    ans.push_back("FizzBuzz");
                } else {
                    ans.push_back("Fizz");
                }
                
            } else if (i%5 == 0){
                if(i%3 == 0){
                    ans.push_back("FizzBuzz");
                } else {
                    ans.push_back("Buzz");
                }
            } else {
                ans.push_back(to_string(i));
            }
        }
        return ans;
    }
};
